controllers/util/solr_util.go

```diff
@@ -266,6 +270,9 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 	solrHostName := solrCloud.AdvertisedNodeHost("$(POD_HOSTNAME)")
 	solrAdressingPort := solrCloud.NodePort()
 
+	// Solr can take longer than SOLR_STOP_WAIT to run solr stop, give it a few extra seconds before forcefully killing the pod.
+	solrStopWait := terminationGracePeriod - 5
```

Comment:
Check for positive value here, somebody might have set grace period on their cluster to 5s or less.

Reply:
maybe we set a minimum possible value for terminationGracePeriod? Like 10 seconds?

Reply:
Added the minimum as well as a check for a negative value, in case we remove the minimum in the future.